src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -163,9 +163,6 @@ public IReadOnlyList<PackageSpec> GetClosure(string rootUniqueName)
                 throw new ArgumentNullException(nameof(rootUniqueName));
             }
 
-            var projectsByUniqueName = _projects
```

Comment:
It looks previous implementation is check OS specific uniqueness of dictionary key. Does `_projects` satisfy this requirement for different OS like Win, MAC, Linux etc?

Reply:
My understanding is `PackageSpec.RestoreMetadata.ProjectUniqueName` is used as a key when adding items to the dictionary. Hence the casing should be the same.  Unless there is a gap in the test coverage for this code path, CI produced a green build for this change.

https://github.com/NuGet/NuGet.Client/blob/b102b7587f0286d83dad8ee357caded56bd0ab17/src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs#L216-L226

Reply:
> My understanding is PackageSpec.RestoreMetadata.ProjectUniqueName is used as a key when adding items to the dictionary. Hence the casing should be the same.

The fact that the project unique name is used does not, by itself, ensure that the behaviour is the same. Imagine two different projects, both of which have a `ProjectReference` to a third project, but use different casing in their `Include` path. On Linux, this will fail* regardless, but on Windows it could succeed* if all the code that uses the path knows to treat it as case insensitive.

*On Linux, Windows, and Mac, file sysstem can be configured to be either case sensitive or insensitive. It's just the deaults on the various operating systems that are different. Applications like NuGet that try to treat it always being the same are going to have bugs when the customer changes their filesystems' settings.


Having said that if @erdembayar checked the definition/initiaization of `_projects`, he'd see that `PathUtility.GetStringComparerBasedOnOS()` was also used.  This is the reason why the behaviour isn't changing with this code change.

Reply:
Thanks, Andy, for additional information. As Andy pointed out `PathUtility.GetStringComparerBasedOnOS()` invoked when `_projects` was defined. 

https://github.com/NuGet/NuGet.Client/blob/a5d5e3236e86f6c3c09097b723ce2b52852a1752/src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs#L22